pkg/handler/common/provider/kubevirt.go

```diff
@@ -224,3 +232,87 @@ func KubeVirtStorageClassesWithClusterCredentialsEndpoint(ctx context.Context, u
+		if err := json.Unmarshal([]byte(vmiPreset.Spec), &presetSpec); err != nil {
+			log.Logger.Errorf("skipping VMIPreset:%s, parsing preset.Spec failed:%v", vmiPreset.Name, err)
+			continue
+		}
+
+		cpu, memory, err := GetKubeVirtPresetResourceDetails(presetSpec)
+		if err != nil {
+			log.Logger.Errorf("skipping VMIPreset:%s, fetching presetResourceDetails failed:%v", vmiPreset.Name, err)
+			continue
+		}
+
+		if handlercommon.FilterCPU(int(cpu.AsApproximateFloat64()), quota.MinCPU, quota.MaxCPU) && handlercommon.FilterMemory(int(memory.Value()/(1<<30)), quota.MinRAM, quota.MaxRAM) {
+			filteredVMIPresets = append(filteredVMIPresets, vmiPreset)
+		}
+	}
+	return filteredVMIPresets
+}
+
+// GetKubeVirtPresetResourceDetails extracts cpu and mem resource requests from the kubevirt preset
+// for CPU, take the value by priority:
+// - check if spec.cpu is set, if socket and threads are set then do the calculation, use that
+// - if resource request is set, use that
+// - if resource limit is set, use that
+// for memory, take the value by priority:
+// - if resource request is set, use that
+// - if resource limit is set, use that.
+func GetKubeVirtPresetResourceDetails(presetSpec kubevirtv1.VirtualMachineInstancePresetSpec) (resource.Quantity, resource.Quantity, error) {
+	var err error
+	// Get CPU
+	cpuReq := resource.Quantity{}
+
+	if presetSpec.Domain.CPU != nil {
+		if !presetSpec.Domain.Resources.Requests.Cpu().IsZero() || !presetSpec.Domain.Resources.Limits.Cpu().IsZero() {
+			return resource.Quantity{}, resource.Quantity{}, errors.New("should not specify both spec.domain.cpu and spec.domain.resources.[requests/limits].cpu in VMIPreset")
+		}
+		cores := presetSpec.Domain.CPU.Cores
+		if cores == 0 {
+			cores = 1
+		}
+		// if threads and sockets are set, calculate VCPU
+		threads := presetSpec.Domain.CPU.Threads
+		if threads == 0 {
+			threads = 1
+		}
+		sockets := presetSpec.Domain.CPU.Sockets
+		if sockets == 0 {
+			sockets = 1
+		}
+
+		cpuReq, err = resource.ParseQuantity(strconv.Itoa(int(cores * threads * sockets)))
+		if err != nil {
+			return resource.Quantity{}, resource.Quantity{}, fmt.Errorf("error parsing calculated KubeVirt VCPU: %w", err)
+		}
+	} else {
+		if !presetSpec.Domain.Resources.Requests.Cpu().IsZero() {
+			cpuReq = *presetSpec.Domain.Resources.Requests.Cpu()
+		}
+		if !presetSpec.Domain.Resources.Limits.Cpu().IsZero() {
+			cpuReq = *presetSpec.Domain.Resources.Limits.Cpu()
+		}
+	}
+
+	// get MEM
+	memReq := resource.Quantity{}
+	if presetSpec.Domain.Resources.Requests.Memory().IsZero() && presetSpec.Domain.Resources.Limits.Memory().IsZero() {
+		return resource.Quantity{}, resource.Quantity{}, errors.New("spec.domain.resources.[requests/limits].memory must be set in VMIPreset")
+	}
+	if !presetSpec.Domain.Resources.Requests.Memory().IsZero() {
+		memReq = *presetSpec.Domain.Resources.Requests.Memory()
+	}
+	if !presetSpec.Domain.Resources.Limits.Memory().IsZero() {
```

Comment:
so the memory can be set through limits in the end? There wont be an error?

Reply:
It depends on the right QoS class. We plan to iterate over this implementation. The same for dedicated cpu placement. For now it's ok.